### PR TITLE
[Fixes #7482]  Implement API v2 endpoints to complete the support for filtering

### LINKED ISSUE #7482: Implement API v2 endpoints to complete the support for filtering
## Expected Behavior
To have the following endpoints in V2 
 - categories
 - keywords
 - regions
 - owners

## Actual Behavior
The new API v2 has only implemented the (new) [resource_types](https://docs.geonode.org/en/master/api/V2/#get--api-v2-resources-resource_types-) endpoint, but we lack the followin:

 - categories
 - keywords
 - regions
 - owners

## Steps to Reproduce the Problem

  1.https://docs.geonode.org/en/master/api/V2 and view available endpionts

## Specifications

  - GeoNode version: master
  - Installation method (manual, GeoNode Docker, SPCGeoNode Docker):
  - Platform:
  - Additional details:

### PR DESCRIPTION
**Pull request by marthamareal**
<Include a few sentences describing the overall goals for this Pull Request>
References  #7482

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [x] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [x] The issue connected to the PR must have Labels and Milestone assigned
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [x] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates
- [ ] Commits adding **new texts** do use gettext and have updated .po / .mo files (without location infos)

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

File: geonode/base/utils.py [hunk 1]
```diff
@@ -141,6 +142,24 @@ def build_absolute_uri(url):
     return url
 
 
+def get_resources(user):
```
Comment (giohappy) [inline geonode/base/utils.py:145]: @marthamareal please either use the already available [filter](https://github.com/GeoNode/geonode/blob/cf0142a57f1fa6c59ee6d4b186d7bf8f0506bc07/geonode/base/api/permissions.py#L150) defined in ResourceBasePermissionsFilter, or make that filter use this method. Please notice that the filter_queryset does something more, for example it checks the SKIP_PERMS_FILTER settings.

**Review by giohappy — CHANGES_REQUESTED** [review]

File: geonode/base/utils.py (hunk 1 above)
Comment (marthamareal) [reply geonode/base/utils.py]: I have updated this @giohappy

**Review by afabiani — APPROVED** [review]

**Comment by afabiani** [comment]
Failing test on upload is not related to this PR. The rest is good.

@marthamareal we will need to update the APIs documentation too. Let's trak the documentation PR here as soon as it will be ready, ok?

**Comment by giohappy** [comment]
Issue related to doc https://github.com/GeoNode/documentation/issues/99

**Pull request merged**